tuxemon/technique.py

```diff
@@ -297,25 +297,32 @@ def changestats(self, user: Monster, target: Monster) -> EffectResult:
         '''
         statsmaster = [self.statspeed, self.stathp, self.statarmour, self.statmelee, self.statranged, self.statdodge]
         statslugs = ['speed', 'hp', 'armour', 'melee', 'ranged', 'dodge']
+        newstatvalue = 0
         for stat, slug in zip(statsmaster, statslugs):
             if not stat:
                 continue
-            value = stat['value']
-            dividing = stat['dividing']
+            minvalue = stat['minvalue']
+            maxvalue = stat['maxvalue']
+            iterator = stat['iterator']
             override = stat['overridetofull']
             basestatvalue = getattr(target, slug)
-            if value > 0 and override == False:
-                if dividing == False:
-                    newstatvalue = basestatvalue + value
-                else:
-                    newstatvalue = basestatvalue // value
-                success = True
+            if maxvalue:
+                value = random.randint(minvalue,maxvalue)
+            else:
+                value = minvalue
+            if value > 0 and  override == False:
+                ops_dict = {
+                "+": operator.add,
+                "-": operator.sub,
+                "*": operator.mul,
+                "/": operator.floordiv,
+            }
+
+                newstatvalue = ops_dict[operator](basestatvalue, value)
```

Comment:
I have just noticed that there is a name clash between the variable `operator` and the module of the same name. The variable should be `operation` instead.

Reply:
fixed